### PR TITLE
Changes the rules about whether Elide can paginate to be less restric…

### PR DESCRIPTION
**Pull request by aklish**
…tive for user checks

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: elide-core/src/main/java/com/yahoo/elide/parsers/expression/CanPaginateVisitor.java [hunk 1]
```diff
@@ -127,67 +127,101 @@
...
+    public CanPaginateVisitor(EntityDictionary dictionary, RequestScope scope) {
         this.dictionary = dictionary;
+        this.scope = scope;
     }
 
-
     @Override
-    public Boolean visitNOT(ExpressionParser.NOTContext ctx) {
-        return visit(ctx.expression());
+    public PaginationStatus visitNOT(ExpressionParser.NOTContext ctx) {
```
Comment (wcekan) [inline elide-core/src/main/java/com/yahoo/elide/parsers/expression/CanPaginateVisitor.java]: Does this need to negate CAN_PAGINATE/CAN_PAGINATE as well?

**Comment by DeathByTape** [comment]
👍

**Comment by wcekan** [comment]
👍

**Pull request merged**